android/src/main/java/com/onegini/mobile/sdk/reactnative/handlers/registration/RegistrationRequestHandler.kt

```diff
@@ -0,0 +1,30 @@
+package com.onegini.mobile.sdk.reactnative.handlers.registration
+
+import com.onegini.mobile.sdk.android.handlers.request.OneginiBrowserRegistrationRequestHandler
+import com.onegini.mobile.sdk.android.handlers.request.callback.OneginiBrowserRegistrationCallback
+import android.net.Uri
+
+class RegistrationRequestHandler : OneginiBrowserRegistrationRequestHandler {
+  private var callback: OneginiBrowserRegistrationCallback? = null
+  private var eventEmitter = RegistrationEventEmitter()
+  /**
+   * Finish registration action with result from web browser
+   */
+  fun handleRegistrationCallback(uri: Uri?) {
+      callback?.handleRegistrationCallback(uri)
+      callback = null
```

Comment:
Have you contacted @aleknow on this? Now when I think of it, is there a chance that on failed `callback?.handleRegistrationCallback(uri)`, `onRegistrationCanceled()` won't be called? I assume that `onRegistrationCanceled()` will force `startRegistration()` to be called, filling back the `callback` variable. If that won't happen, and `handleRegistrationCallback()` will be called for the second time, it will result in nothing happening.

Reply:
I believe you are correct here.
If the `callback?.handleRegistrationCallback(uri)` somehow fails we will indeed set the callback to null but the registration will still be ongoing. That means that we won't be able to cancel the registration and we will be stuck in a state where we can not register and can not cancel. 
I will consult with Alek to see what we can do about this, we need to somehow see if the `callback?.handleRegistrationCallback(uri)` succeeds but this function returns void. 
I 
For your other point `onRegistrationCanceled()` will not force `startRegistration()` this function just seems to have a weird name and should probably just be named `cancelRegistration()` I think that is what is causing the confusion.

Reply:
Ok, so I tested this and talked to @aleknow about it. 
It seems that when `callback?.handleRegistrationCallback(uri)` fails, it will cancel the registration so we can safely set the callback to null.